x-pack/plugin/search-business-rules/src/main/java/org/elasticsearch/xpack/searchbusinessrules/PinnedQueryBuilder.java

```diff
@@ -105,13 +104,21 @@ private Item(String id) {
          * Read from a stream.
          */
         public Item(StreamInput in) throws IOException {
-            index = in.readString();
+            if (in.getTransportVersion().onOrAfter(OPTIONAL_INDEX_IN_DOCS_VERSION)) {
+                index = in.readOptionalString();
+            } else {
+                index = in.readString();
+            }
             id = in.readString();
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            out.writeString(index);
+            if (out.getTransportVersion().onOrAfter(OPTIONAL_INDEX_IN_DOCS_VERSION)) {
+                out.writeOptionalString(index);
+            } else {
+                out.writeString(index);
```

Comment:
This will throw an NPE. It is possible that the coordinator is upgraded to the latest version of Elasticsearch but the relevant data nodes are not. Consequently, if the user does not provide an `_index`, the coordinator will accept the call, attempt to serialize, and an NPE will throw. 

We should instead be nice here and throw a better error message.

Reply:
Good catch - I've included an `IllegalArgumentException` that hopefully makes sense 👍